packages/react-mui-audit-log-viewer/src/AuditLogViewer.tsx

```diff
@@ -15,43 +15,82 @@ import { DEFAULT_AUDIT_SCHEMA, useSchema } from "./hooks/schema";
 import { PublicAuditQuery } from "./types/query";
 import { useAuditSearchError } from "./hooks/query";
 
+/** Properties related to the behavior and use of the AuditLogViewer. */
 export interface AuditLogViewerProps<Event = Audit.DefaultEvent> {
+  /** The default initial search query */
   initialQuery?: string;
 
+  /** Called when the user performs a search. Should make a call to the Audit Service /search endpoint proxied through your application server */
   onSearch: (body: Audit.SearchRequest) => Promise<Audit.SearchResponse>;
+
+  /** Prevent the search input to auto search on change. If false will only search when the "Search" button is clicked or if the "Enter" key is typed while focused on the search input */
   searchOnChange?: boolean;
+
+  /** Trigger search on component mount. By default searchOnChange triggers a search, but if it is disabled then searchOnFilterChange can be enabled */
   searchOnFilterChange?: boolean;
+
+  /**
+   * @hidden
+   */
   searchOnMount?: boolean;
```

Comment:
/** Flag to control is onSearch callback should be trigger on component mount. Set to false to prevent search on component load. Defaults true. */